src/lib/statuses.ts

```diff
@@ -10,14 +13,36 @@ import {
   VOWEL_STATUS_HIRAGANA,
 } from '../constants/strings'
 
-export type CharStatus = 'absent' | 'vowel' | 'consonant' | 'present' | 'close' | 'correct'
+export type CharStatus =
+  | 'absent'
+  | 'vowel'
+  | 'consonant'
+  | 'present'
+  | 'close'
+  | 'correct'
+
+const charStatusRanking = {
+  correct: 5,
+  close: 4,
+  present: 3,
```

Comment:
There is a decision here as to whether close has higher priority than present. Currently the code priorities close over present, evidenced by the line

```
if (splitSolution.includes(letter) && !['close', 'correct'].includes(charObj[toHiragana(letter)])) {
        //make status present
        return (charObj[toHiragana(letter)] = 'present')
}
```

so I have left as is. I think think this makes sense as is, since 'close' gives a maximum of 2 other characters it could be (e.g. close は -> ば or ぱ) whereas present gives 3 possible places, so close is more specific.